src/deploy/functions/deploy.ts

```diff
@@ -11,9 +11,32 @@ import * as gcs from "../../gcp/storage";
 import * as gcf from "../../gcp/cloudfunctions";
 import * as gcfv2 from "../../gcp/cloudfunctionsv2";
 import * as backend from "./backend";
+import crypto from "crypto";
 
 setGracefulCleanup();
 
+async function generateSourceHash(source: args.Source): any {
+  const hash = crypto.createHash("sha256");
+  const sourceFile = source.functionsSourceV2 || source.functionsSourceV1;
+  // Hash the contents of the source file
+  if (sourceFile) {
+    const readStream = fs.createReadStream(sourceFile);
+    readStream.pipe(hash);
+    await new Promise((resolve, reject) => {
+      hash.on("end", () => resolve(hash.read()));
+      readStream.on("error", (err) => reject(err));
+    });
+  }
+
+  // Hash any possible .env additions
+  hash.push({
+    ...process.env,
```

Comment:
Is this... all the environment variables of the shell? That's... rather a lot. Is there a way to pull the is it worth trying to pull just the relevant ones?

This worries me because even in my SSH'd shell, I have some variables that look like they change every time I log in which are never going to be relevant to deploying my functions. However, that would prevent this from ever working because my shell changes all the time. What do you think?

Reply:
Shoot, I just tested it and you're right! (I thought it was limited to args passed, which would have included the .env